Signal/src/ViewControllers/ConversationView/ConversationInputTextView.m

```diff
@@ -142,7 +142,15 @@ - (void)updateTextContainerInset
         }
     }
 
-    self.textContainerInset = UIEdgeInsetsMake(7.f, leftInset, 7.f, rightInset);
+    // Check the system font size and increase text inset accordingly
+    // to keep the text vertically centered
+    CGFloat currentFontSize = self.font.pointSize;
+    // Default body font size is 17 according to the HIG (Typography)
+    CGFloat systemDefaultFontSize = 17.f;
+    CGFloat ourDefaultInset = 7.f;
+    CGFloat topInset = ourDefaultInset + (systemDefaultFontSize > currentFontSize ? systemDefaultFontSize - currentFontSize : 0);
+    CGFloat bottomInset = systemDefaultFontSize > currentFontSize ? topInset - 1 : topInset;
+    self.textContainerInset = UIEdgeInsetsMake(topInset, leftInset, bottomInset, rightInset);
```

Comment:
Does this work correctly when you increase your font size above the default? It seems like the math will do weird things when `currentFontSize` is greater than `systemDefaultFontSize`

Reply:
The intention is to do nothing special when the font size is equal to or larger than the default. When it is smaller than default, inset increases to keep the line pushed down more; otherwise the default inset is too small the line gets closer to the top side. On the other hand, if the font is larger than the default, the input box gets larger to accommodate the larger texts, so the insets don't have to change.

Reply:
Ah, got it. I misread this initially.